src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -427,102 +425,24 @@ protected function initializeContainer()
         $class = $this->getContainerClass();
         $cacheDir = $this->warmupDir ?: $this->getCacheDir();
         $cache = new ConfigCache($cacheDir.'/'.$class.'.php', $this->debug);
-        $cachePath = $cache->getPath();
 
         // Silence E_WARNING to ignore "include" failures - don't use "@" to prevent silencing fatal errors
         $errorLevel = error_reporting(E_ALL ^ E_WARNING);
 
-        try {
-            if (is_file($cachePath) && \is_object($this->container = include $cachePath)
-                && (!$this->debug || (self::$freshCache[$cachePath] ?? $cache->isFresh()))
```

Comment:
`self::$freshCache` was used here. Not using it anymore in your new code means that you reverted a feature. That's a no-go for a refactoring.

Reply:
by each request and bootstrapping kernel, `freshCache` is empty so `cache->isFresh()` called and set `freshCache` to be true...could you explain to me what happened if it removes? and what is that feature?

Reply:
this happens when running tests, where you would manage lots of requests in the same process (and also when using something like php-pm)

Reply:
if you remove it, testsuites become a lots slower again (especially if you don't disable debug mode when running tests).

Reply:
thank you, you are right. I reverted this commit.